lib/index.js

```diff
@@ -267,11 +274,21 @@ export default {
             range = Helpers.generateRange(textEditor, line);
           }
 
+          // avoid duplicated issues
+          if (fingerprints.has(issue.fingerprint)) {
+            return;
+          }
+          fingerprints.add(issue.fingerprint);
+
           linterResults.push({
-            type: 'Warning',
-            text: issue.description,
-            filePath,
-            range,
+            severity: mapSeverity[issue.severity] || 'warning',
+            excerpt: `${issue.engine_name.toUpperCase()} [${issue.check_name}]: ${issue.description}`,
```

Comment:
So _personally_ I would add all that info as well, however a lot of people complain when "extra" stuff is added like that, which is why I was asking if that matched how `codeclimate` itself output the information.

> What do you mean? How would you do that?

In a few other providers things like the rule ("check") name are put behind optional flags, allowing people to hide them if they don't feel the need for them. So in implementing that you would just do an `if` on the state of that flag option.

Hold off on implementing all that though since I'd probably be fine with all the information there, but you might want to put the `check_name` at the end of the line since the message (tied with the location) is probably sufficient to fix the problem.